packages/override-tools/src/CrossProcessLock.ts

```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ *
+ * @format
+ */
+
+import * as net from 'net';
+
+/**
+ * CrossProcessLock implements a non-polling cross-process asynchronus barrier.
```

Comment:
>CrossProcessLock [](start = 3, length = 16)

Just another thought I had after our conversation. If P0 can just execute P0p1 and P0p2 sequentially. It wouldn't try to execute them in parallel in the first place. P0p1 and P0p2 tasks are independent but sequential.